src/output-json-alert.c

```diff
@@ -169,6 +171,33 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
     jb_close(js);
 }
 
+static void AlertJsonReference(
+        AlertJsonOutputCtx *json_output_ctx, const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    const DetectReference *kv = pa->s->references;
+    jb_open_array(jb, "references");
+    while (kv) {
+        const size_t key_size = MIN(kv->key_len, REFERENCE_SYSTEM_NAME_MAX);
+        const size_t ref_size = MIN(kv->reference_len, REFERENCE_CONTENT_NAME_MAX);
+        const size_t size_needed = key_size + ref_size + 1;
+        char kv_store[size_needed];
+        // When the reference begins with the key, don't append the key
+        // e.g., the reference URI may already contain the scheme son
+        // don't append https:// again
+        if (strncmp(kv->reference, kv->key, key_size) == 0)
```

Comment:
can we be a 100% sure here that strlen(reference) >= key_size? Or will it bounds check correctly?

Reply:
Good catch -- I'll ensure that reference's size is GTE to the size of the key.